frontend/awx/resources/templates/WorkflowVisualizer/hooks/useWorkflowVisualizerToolbarActions.tsx

```diff
@@ -5,19 +5,69 @@ import {
   CheckCircleIcon,
   CloseIcon,
   CompressAltIcon,
+  EllipsisVIcon,
   ExpandAltIcon,
   PlusCircleIcon,
+  TrashIcon,
 } from '@patternfly/react-icons';
-import { Badge, Button, ToolbarGroup, ToolbarItem } from '@patternfly/react-core';
+import {
+  Badge,
+  Button,
+  Dropdown,
+  DropdownItem,
+  DropdownList,
+  MenuToggle,
+  ToolbarGroup,
+  ToolbarItem,
+} from '@patternfly/react-core';
 import { WorkflowNode } from '../../../../interfaces/WorkflowNode';
-import { usePageNavigate } from '../../../../../../framework';
+import { useBulkConfirmation, usePageNavigate } from '../../../../../../framework';
 import { AwxRoute } from '../../../../AwxRoutes';
+import { getItemKey, requestDelete } from '../../../../../common/crud/Data';
+import { stringIsUUID } from '../../../../common/util/strings';
 
 export function useWorkflowVisualizerToolbarActions(nodes: WorkflowNode[]) {
   const { t } = useTranslation();
   const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const [isKebabOpen, setIsKebabOpen] = useState<boolean>(false);
   const pageNavigate = usePageNavigate();
   const params = useParams<{ id: string }>();
+  const bulkAction = useBulkConfirmation<WorkflowNode>();
+  const handleDeleteAllNodes = useCallback(
+    (workflowNodes: WorkflowNode[]) => {
+      return bulkAction({
+        title: t('Delete all workflow nodes'),
```

Comment:
```suggestion
        title: t('Remove all nodes'),
```
https://www.sketch.com/s/46cfe983-742c-4843-a462-6bd7bc04d8e1/a/oM8VKvL